File: linear-search/linear-search.clj
(defn linear-search [arr target]
  (loop [i 0]
    (cond
      (>= i (count arr)) -1
      (= (nth arr i) target) i
      :else (recur (inc i)))))

(defn linear-search-recursive [arr target index]
  (cond
    (>= index (count arr)) -1
    (= (nth arr index) target) index
    :else (linear-search-recursive arr target (inc index))))

(println (linear-search [1 2 3 4 5] 3))
(println (linear-search-recursive [1 2 3 4 5] 3 0))

File: linear-search/linear-search.js
function linearSearch(arr, target) {
    for (let i = 0; i < arr.length; i++) {
        if (arr[i] === target) return i;
    }
    return -1;
}

function linearSearchRecursive(arr, target, index = 0) {
    if (index >= arr.length) return -1;
    if (arr[index] === target) return index;
    return linearSearchRecursive(arr, target, index + 1);
}

console.log(linearSearch([1, 2, 3, 4, 5], 3));
console.log(linearSearchRecursive([1, 2, 3, 4, 5], 3));